neural_compressor/torch/quantization/quantize.py

```diff
@@ -12,46 +12,57 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import Any, Callable
+from typing import Any, Callable, Dict, Tuple
 
 import torch
 
-from neural_compressor.common.base_config import BaseConfig
+from neural_compressor.common.base_config import BaseConfig, ComposableConfig, registered_configs
 from neural_compressor.common.logger import Logger
-from neural_compressor.common.utility import RTN_WEIGHT_ONLY_QUANT
-from neural_compressor.torch.quantization.config import parse_config_from_dict
-from neural_compressor.torch.utils import algos_mapping
+from neural_compressor.torch.quantization.config import FRAMEWORK_NAME
+from neural_compressor.torch.utils import algos_mapping, get_model_info
 
 logger = Logger().get_logger()
 
 
+def need_apply(configs_mapping: Dict[Tuple[str, callable], BaseConfig], algo_name):
+    return any(config.name == algo_name for config in configs_mapping.values())
+
+
 def quantize(
-    model: torch.nn.Module, quant_config: BaseConfig, calib_func: Callable = None, calib_func_arg: Any = None
+    model: torch.nn.Module,
+    quant_config: BaseConfig,
+    calib_dataloader=None,
```

Comment:
Hi @ftian1, please be aware that I added `calib_dataloader` as a new argument for `quantize`, as GPTQ needs a dataloader for calibration.

Reply:
we need use run_fn / run_args to do that but not this calib_dataloader

Reply:
removed and renamed by [7596137](https://github.com/intel/neural-compressor/pull/1408/commits/759613763b806f58edfd8ca6f4381ede69c0b505) and [1f02346](https://github.com/intel/neural-compressor/pull/1408/commits/1f02346ec24f94abfe44c3bc2050c0a5bc1afbfb)

Reply:
refactored gptq to use `run_fn` and `run_args` by [42a07d9](https://github.com/intel/neural-compressor/pull/1408/commits/42a07d958fbd22efc8dea680f144b5a81d9f0c08)